src/lib/tree/tree.md

```diff
@@ -0,0 +1,50 @@
+The `mat-tree` provides a Material Design styled tree that can be used to display hierarchy
+data.
+
+This tree builds on the foundation of the CDK tree and uses a similar interface for its
```

Comment:
I would mention the concept of flat vs. nested trees here, as early as possible.

Reply:
Added flat tree and nested tree part. Please take a look. Thanks!